app/controllers/avo/application_controller.rb

```diff
@@ -142,7 +142,8 @@ def set_related_resource
     end
 
     def set_record
-      @record = @resource.find_record(params[:id], query: model_scope, params: params)
+      id = @resource.model_class.primary_key.is_a?(Array) ? params.extract_value(:id) : params[:id]
+      @record = @resource.find_record(id, query: model_scope, params:)
```

Comment:
Let's keep the ` params[:id]` call for versions that doesn't have the `extract_value` method
```suggestion
id = if @resource.model_class.primary_key.is_a?(Array) && params.respond_to?(:extract_value)
  params.extract_value(:id)
else
  params[:id]
end

@record = @resource.find_record(id, query: model_scope, params:)
```

Reply:
When I opened the PR I was thinking in checking the Rails version, your suggestion is much better. Thanks.